numpy/lib/tests/test_nanfunctions.py

```diff
@@ -83,6 +83,17 @@ def test_result_values(self):
             res = nf(_ndat, axis=1)
             assert_almost_equal(res, tgt)
 
+    def test_initializer(self):
+        arr1 = np.array([np.nan] * 9).reshape(3, 3)
+        arr2 = arr1.copy()
+        arr2[1] = 1
+
+        assert_equal(np.nanmin(arr1, initializer=5), 5.0)
+        assert_equal(np.nanmax(arr1, initializer=6), 6.0)
+
+        assert_equal(np.nanmin(arr2, initializer=5), 1.0)
+        assert_equal(np.nanmax(arr2, initializer=6), 6.0)
```

Comment:
Can you add some tests of `np.nanmax([], initializer=np.nan)`, `np.nanmax([np.nan], initializer=np.nan)`, and `np.nanmax([np.nan], initializer=0)`?

Reply:
What me and @mhvk differ on is the behavior of `np.nanmax([0], initializer=np.nan)`. I believe this should be zero, he says it should err or warn; and return `nan`.

Reply:
And sure, I can add those tests. Not immediately though. It'd be best if you could settle this debate first.

Reply:
Just to be clear, in my ideal world `initializer=np.nan` just raises an exception. I haven't checked what it actually does if you remove the turn-it-into-0 stanza.

Reply:
Perhaps we should leave the nanfunctions to another PR?

Reply:
> Perhaps we should leave the nanfunctions to another PR?

I'm okay with that.

> Just to be clear, in my ideal world initializer=np.nan just raises an exception. I haven't checked what it actually does if you remove the turn-it-into-0 stanza.

It will make everything `nan`.

Reply:
> It will make everything `nan`.

Not my ideal world, but I think preferable to silently changing it. But maybe it is indeed best to just move the `nanfunctions` to another PR, so we can merge this one.